doc/source/whatsnew/v0.23.0.txt

```diff
@@ -119,6 +119,47 @@ Current Behavior
 
     s.rank(na_option='top')
 
+
+.. _whatsnew_0230.enhancements.assign_dependent:
+
+``.assign()`` accepts dependent arguments
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :func:`DataFrame.assign()` now accepts dependent kwargs for python version later than 3.6 (see also `PEP 468
+<https://www.python.org/dev/peps/pep-0468/>`_). Now the keyword-value pairs passed to `.assign()` may depend on their predecessors if the values are callables. (:issue:`14207`)
+
+.. ipython:: python     
+
+    df = pd.DataFrame({'A': [1, 2, 3]})
+    df
+	df.assign(B=df.A, C=lambda x:x['A']+ x['B'])
+
+.. warning::
+
+This may subtly change the behavior of your code when you're
```

Comment:
add this warning to the main docs as well in dsintro.rst

Reply:
What are the Main docs? Docstring of the assign function?

Reply:
http://pandas-docs.github.io/pandas-docs-travis/dsintro.html#assigning-new-columns-in-method-chains

That's in doc/source/dsintro.rst.